dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -0,0 +1,173 @@
+        final String siteName = null == request.getParameter(Host.HOST_VELOCITY_VAR_NAME) ?
+                request.getServerName() : request.getParameter(Host.HOST_VELOCITY_VAR_NAME);
+        final Host site = this.hostWebAPI.resolveHostName(siteName, user, RESPECT_FE_ROLES);
+
+        final HTMLPageAsset page = (HTMLPageAsset) this.htmlPageAssetAPI.getPageByPath(pageUri,
+                site, this.languageAPI.getDefaultLanguage().getId(), true);
+        if (isRendered) {
+            try {
+                page.setProperty("rendered", VelocityUtil.mergeTemplate("/live/" + page
+                        .getIdentifier() + "_" + page.getLanguageId() + ".dotpage",
+                        velocityContext));
+            } catch (Exception e) {
+                throw new DotDataException(String.format("Page '%s' could not be rendered via " +
+                        "Velocity.", pageUri), e);
+            }
+        }
+        final Template template = (Template) this.versionableAPI.findLiveVersion(page
+                .getTemplateId(), user, RESPECT_ANON_PERMISSIONS);
+        final List<Container> templateContainers = this.templateAPI.getContainersInTemplate
+                (template, user, RESPECT_FE_ROLES);
+        final Map<String, ContainerView> mappedContainers = new LinkedHashMap<>();
+        for (Container container : templateContainers) {
+            final List<ContainerStructure> containerStructures = this.containerAPI
+                    .getContainerStructures(container);
+            String rendered = null;
+            if (isRendered) {
+                try {
+                    rendered = VelocityUtil.mergeTemplate("/live/" + container.getIdentifier() +
+                            ".container", velocityContext);
+                } catch (Exception e) {
+                    throw new DotDataException(String.format("Container '%s' could not be " +
+                            "rendered via " + "Velocity.", container.getIdentifier()), e);
+                }
+            }
+            mappedContainers.put(container.getIdentifier(), new ContainerView(container,
+                    containerStructures, rendered));
+        }
+
+        TemplateLayout layout = null;
+        if (null != template.getTheme()) {
+            final DotTemplateTool dotTemplateTool = new DotTemplateTool();
+            dotTemplateTool.init(velocityContext);
+            layout = dotTemplateTool.themeLayout(template.getInode());
+        }
+
+        final PageView pageView = new PageView(site, template, mappedContainers, page, layout);
+        return pageView;
+    }
+
+    /**
+     *
+     * @param pageView
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String asJson(final PageView pageView) throws JsonProcessingException {
+        final ObjectWriter objectWriter = JsonMapper.mapper.writer().withDefaultPrettyPrinter();
+        final String json = objectWriter.writeValueAsString(pageView);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "json". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)